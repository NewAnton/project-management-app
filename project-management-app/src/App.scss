.header {
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
}

.main {
  margin: 0 auto;
  max-width: 100vw;
  min-height: calc(100vh - 6.85rem);
  padding: 4.5rem 1rem 1rem 1rem;
  background-image: url('./assets/bg-1.jpg');
  background-size: cover;
  .main__title {
  font-weight: 700;
  color: var(--secondary-color);
  margin-bottom: 2rem;
}
}

.footer {
  background-color: var(--footer-bg);
  border-top: .15rem solid var(--text-color);
}
