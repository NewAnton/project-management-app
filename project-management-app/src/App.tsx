import { Route, Routes, useNavigate } from 'react-router-dom';
import React, { useEffect, useMemo } from 'react';

import { Navigation } from './components/Navigation/Navigation';
import { Main } from 'pages/Main/Main';
import { BoardList } from 'pages/BoardList/BoardList';
import { Board } from 'pages/Board/Board';
import { Profile } from 'pages/Profile/Profile';
import { Task } from 'components/Task/Task';
import { ErrorPage404 } from 'pages/404ErrorPage/404ErrorPage';
import { Footer } from 'components/Footer/Footer';
import { useTypedSelector } from 'hooks/useTypedSelector';
import { checkIsTokenExpired } from 'services/checkIsTokenExpired';
import { SignIn } from 'pages/SignIn/SignIn';
import { SignUp } from 'pages/SignUp/SignUp';

import './App.scss';
import { NewBoard } from 'components/NewBoard/NewBoard';
import { useGetUserByIdQuery } from 'services/kanbanApiUsers';
import { decodeToken } from 'react-jwt';
import { DecodedTokenInterface } from 'types/DecodedTokenInterface';

export function App() {
  const { token } = useTypedSelector((state) => state.globalState);
  const navigate = useNavigate();

  const isTokenExpired = useMemo(() => checkIsTokenExpired(token), [token]);

  const { data } = useGetUserByIdQuery((decodeToken(token) as DecodedTokenInterface)?.id || '', {
    skip: !token,
  });

  useEffect(() => {
    localStorage.setItem('login', data?.login || '');
  }, [data]);

  useEffect(() => {
    if (isTokenExpired) navigate('/');
  }, [isTokenExpired]);

  return (
    <>
      <header className="header">
        <Navigation isTokenExpired={isTokenExpired} />
      </header>
      <main className="main">
        <Routes>
          <Route path="/" element={<Main />} />
          <Route path="/new-board" element={<NewBoard />} />
          <Route path="/board-list" element={<BoardList />} />
          <Route path="/profile" element={<Profile />} />
          <Route path="/board/:id" element={<Board />} />
          <Route path="/task" element={<Task />} />
          <Route path="/sign-in" element={<SignIn />} />
          <Route path="/sign-up" element={<SignUp />} />
          <Route path="*" element={<ErrorPage404 />} />
        </Routes>
      </main>
      <Footer />
    </>
  );
}
