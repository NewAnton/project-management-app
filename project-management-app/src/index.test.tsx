import { App } from './App';
import Index from './index';

jest.mock('react-dom/client', () => ({
  createRoot: jest.fn().mockImplementation(() => ({
    render: jest.fn(),
  })),
}));

describe('Application root', () => {
  test('should render without crashing', () => {
    expect(
      JSON.stringify(Object.assign({}, Index, { _reactInternalInstance: 'censored' }))
    ).toMatchSnapshot();
  });
  test('should render the app inside div which has root id', () => {
    expect(global.document.getElementById('root')).toBeDefined();
  });

  test('should render App component', () => {
    expect(App).toBeDefined();
  });
});
