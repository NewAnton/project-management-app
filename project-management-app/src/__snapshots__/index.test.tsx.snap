// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application root should render without crashing 1`] = `"{\\"_reactInternalInstance\\":\\"censored\\"}"`;
